### PR TITLE
Testingsprint4

### PR DESCRIPTION
**Pull request by shushengfuqin**
### Summary 
Modified some unit test to follow current code

### Details
Add getjson test to test wether it get a feed back.
The main change of this pr is in package.json and some refactoring. 

### How to test?
yarn install --dev
yarn test 
all test shall pass

yarn dev
make sure that the program works the way is was.

### CODE REVIEW

File: src/test/WebglBox.test.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import {render} from "@testing-library/react"
+import React from 'react';
+import WebglBox from '../components/webgl/WebglBox'
+
+test('ApparatusSelectedListBox renders without crashing', ()=>{
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/test/WebglBox.test.tsx:5]: not yet. i'm adding one by one. the version that i have before it's not compatible anymore. thus i have to change the entire config. The main reason of this pr is to make sure that the config that i changed is working also on the computer of u guys.

File: src/test/getimage.test.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import {render, screen} from "@testing-library/react"
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/test/getimage.test.tsx:1]: is going to be delete right after, i'm going to go through a lot of re-organization after this

File: src/test/getjson.test.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import getjsonfromurl from '../util/getjsonfromurl';
+
+global.fetch = jest.fn(() =>
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/test/getjson.test.tsx:3]: this part is mocking the fetch. fetch itself is not being described. Thus I have to mock the fetch in order to use it.

File: src/test/getjson.test.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  Promise.resolve({
+    json: () => Promise.resolve({ test: 100 }),
+  }),
+) as jest.Mock;
+
+test('based on the url, it should return a json object', () => {
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/test/getjson.test.tsx:9]: yes the object is JSON object. I'm not quite sure how to test if the entire JSON is the same.

**Review by Ysabz — APPROVED** [review]
This PR needs a supplementary PR to complete the testing

**Pull request merged**